server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -569,6 +601,33 @@ private void allocateExistingUnassignedShards(RoutingAllocation allocation) {
         }
     }
 
+    /**
+     * Verify if all unassigned shards are allocated by the same allocator, if yes then return the allocator, else
+     * return null
+     * @param allocation {@link RoutingAllocation}
+     * @return {@link ExistingShardsAllocator} or null
+     */
+    private ExistingShardsAllocator verifySameAllocatorForAllUnassignedShards(RoutingAllocation allocation) {
+        // if there is a single Allocator set in Allocation Service then use it for all shards
+        if (existingShardsAllocators.size() == 1) {
+            return existingShardsAllocators.values().iterator().next();
+        }
+        RoutingNodes.UnassignedShards unassignedShards = allocation.routingNodes().unassigned();
+        RoutingNodes.UnassignedShards.UnassignedIterator iterator = unassignedShards.iterator();
+        ExistingShardsAllocator currentAllocatorForShard = null;
+        if (unassignedShards.size() > 0) {
+            ShardRouting shard = iterator.next();
+            currentAllocatorForShard = getAllocatorForShard(shard, allocation);
+            while (iterator.hasNext()) {
+                ExistingShardsAllocator allocatorForShard = getAllocatorForShard(iterator.next(), allocation);
+                if (currentAllocatorForShard.getClass().getName().equals(allocatorForShard.getClass().getName()) == false) {
+                    return null;
```

Comment:
What do you plan to do with `null` if allocator was found different? Shouldn't be possible right given there's no code bug? Do you think `IllegalStateException` would be a better fit?

Reply:
It wont be IllegalStateException with respect to AllocationService since it will fallback to non batched version.
Null is send because we didnt have same Allocators set

probably I will rename it to ```getAndVerifySameAllocatorForAllUnassignedShards```